tests/analyse/test_scenario.py

```diff
@@ -1445,6 +1445,56 @@ def test_stack_datasets_with_alignment(flux_daily, flux_daily_small_dim_diff):
     np.testing.assert_allclose(output_flux, expected_flux)
 
 
+def test_scenario_infer_flux_source_ch4():
```

Comment:
To check - was this removing "waste" so this would be the only ch4 emissions file found and so could infer "anthro" as the source? If so, a nicer test would be to add a single flux file for a different long-lived gas (can create dummy flux data if needed here) and checking this rather than altering what's in the object store as this point.

Reply:
Yes that's the reason for removing 'waste'.

The dummy fixtures used elsewhere in `test_scenario` wouldn't work since the change I made only applies when the `ModelScenario` needs to retrieve the flux; if I mock `get_flux`, I would need to validate the input somehow, since this should test a set of arguments that can actually be used to construct a model scenario.

If I change the species, I need to add matching boundary conditions, since `domain` is required for flux, and if `domain` is not `None`, then `ModelScenario.__init__` will try to fetch boundary conditions.


I decided that deleting a source would be fine since the last test clears the store anyway. (Admittedly, the last test tripped me up because I tried to add a new test after it and didn't realise that the store had been cleared at that point.)

Reply:
So you wouldn't need to add matching boundary conditions as you don't need to add boundary conditions to a ModelScenario instance at all. It can try and fail to find boundary conditions and that should be fine (if it's not that's a different problem).

Reply:
In principle, you could also just create an empty `ModelScenario` instance and test the `ModelScenario.add_flux` method directly since this is the function that has been updated.

Reply:
Okay thanks, I'll try those ideas.

Reply:
Did you have a look into the other ideas for this here? This test does seem to check what you want it to check but I'm just considering if this could have unintended consequences for other tests later on if there's an expectation the "waste" emissions are there and they're not.